### ISSUE TITLE
[HOLD #333755] [$500] [MEDIUM] Distance: Create a fork for react-beautiful-dnd and remove patch

### ISSUE THREAD

**Issue by hayata-suenaga** [Expensify/App#28797]
To fix some bugs found in `react-beautiful-dnd`, we created a [patch file](https://github.com/Expensify/App/blob/281e3e7a7e048490f869f0412fb82d303557dfc1/patches/react-beautiful-dnd%2B13.1.1.patch).

There was [a discussion](https://github.com/Expensify/App/pull/26307#discussion_r1311896106) on whether we should consider creating a fork for better maintainability.

There is a strong argument that a fork should be created instead of leaving the patch file. Create a fork.

cc: @kosmydel @fabioh8010 @blazejkustra 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01be5681d4b0e2511a</li>
        <li>Upwork Job ID: 1710111278084698112</li>
        <li>Last Price Increase: 2023-10-16</li>
    </ul>
</details>

**Comment by fabioh8010** [comment]
FYI I will be on vacations starting tomorrow for the next two weeks and will return on Oct 23, but I will leave my thoughts here.

First of all, patches are intended to be temporary fixes until you have a newer release of the library you are patching with the proper fix. In this case, we face the first problem here: [The authors of the library made it clear that they won't be addressing fixes / PRs unless is a security issue](https://github.com/atlassian/react-beautiful-dnd#%EF%B8%8F-maintenance--support).

> This library continues to be relied upon heavily by Atlassian products, but we are focused on other priorities right now and have no current plans for further feature development or improvements.
> 
> It will continue to be here on GitHub and we will still make critical updates (e.g. security fixes, if any) as required, but will not be actively monitoring or replying to issues and pull requests.
> 
> We recommend that you don’t raise issues or pull requests, as they will not be reviewed or actioned until further notice.

So we can assume that this patch file has a high probability of staying here forever in our repo if we continue to use this library.

Then the second problem comes, that is simply hard to make changes and manage patch files. It's difficult to understand why they are there and what they are fixing exactly. Currently we have means to separate patches fixes by using different files (e.g. the react-native patch files we have), but they need extra careful when managing and creating new ones, which increases complexity.

I understand that managing a separate fork comes with an overhead to Expensify, and I'm totally fine when you are dealing with temporary situations, but this one is different by the reasons I mentioned first.

**Comment by hayata-suenaga** [comment]
# Please HOLD until we create a fork repository

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Create a fork for react-beautiful-dnd and remove patch

### What is the root cause of that problem?
It is a improvement

### What changes do you think we should make in order to solve the problem?

1. We need to apply all changes in the patch file to fork repository 
2. Remove the patch of this lib on patch file
3. Add the fork to package.json
4. npm install to load this lib
5. Test again to make sure it works well

**Comment by trjExpensify** [comment]
@hayata-suenaga this issue has `no status` in the wave5 project. Should it be CRITICAL, HIGH, MEDIUM or LOW? Thanks!

**Comment by hayata-suenaga** [comment]
don't know why the title wasn't updated to put the price. updating the title

**Comment by nebiyuelias1** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We're using a package called [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd) that is no longer being maintained. To fix problems, we've created a patch and now it's decided to maintain the package ourselves. Therefore, we've forked the repo.

### What is the root cause of that problem?
The package is no longer being maintained. It says on their page:

> This library continues to be relied upon heavily by Atlassian products, but we are focused on other priorities right now and have no current plans for further feature development or improvements.
> 
> It will continue to be here on GitHub and we will still make critical updates (e.g. security fixes, if any) as required, but will not be actively monitoring or replying to issues and pull requests.
> 
> We recommend that you don’t raise issues or pull requests, as they will not be reviewed or actioned until further notice.

### What changes do you think we should make in order to solve the problem?
We need to apply the patch to our forked repo. We can apply the .patch file using the `git apply` command:
```
git apply react-beautiful-dnd+13.1.1.patch
```
We can then create a pr to the forked repo. After the pr is merged we can use the forked repo in our Expensify project.

### What alternative solutions did you explore? (Optional)

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Create a fork for react-beautiful-dnd and remove patch
### What is the root cause of that problem?

NA / We're forking.

### What changes do you think we should make in order to solve the problem?

1. Clone the forked repo.
2. Apply patch (from App repo) to the forked repo, so that forked repo would contain the changes we wanted.
> We can use git apply,  https://github.com/ds300/patch-package#notes
> Fork means expensify.
4. Push to NPM and update the dependencies.
    1. Remove the existing dnd package.
    2. Remove patch.
    3. Install new package (forked one)

5. Update codebase to use new package (forked) inside the `DraggableList` component folder.

> Or maybe we can create module alias, this is useful is any of packages from `node_modules` is using `react-beautiful-dnd` inside one of their dependencies. We're doing same with `react-native`
https://github.com/Expensify/App/blob/5005161e0a4cbb0fa9e6afb693fa0690cac994c7/config/webpack/webpack.common.js#L186



### What alternative solutions did you explore? (Optional)
NA

**Comment by narefyev91** [comment]
Proposal from @DylanDylann looks good to me https://github.com/Expensify/App/issues/28797#issuecomment-1749951174
🎀 👀 🎀 C+ reviewed

**Comment by DylanDylann** [comment]
@hayata-suenaga I am working on this, but encountered an issue when applying the fork repo "react-beautiful-dnd"

**Comment by hayata-suenaga** [comment]
The PR is being reviewed

**Comment by hayata-suenaga** [comment]
The PR is still being reviewed

**Comment by hayata-suenaga** [comment]
[pinged the C+](https://github.com/Expensify/react-beautiful-dnd/pull/1#issuecomment-1785595194) if they're able to test the PR

**Comment by hayata-suenaga** [comment]
The PR to edit the fork was merged.

@DylanDylann can you also make a PR that adds the forked repo to App and replace the current import of the library to the forked repo?

**Comment by DylanDylann** [comment]
@hayata-suenaga Yeah. But can you help publish this fork package to npm?

**Comment by DylanDylann** [comment]
@hayata-suenaga friendly BUMP in case you miss it https://github.com/Expensify/App/issues/28797#issuecomment-1791745042

**Comment by hayata-suenaga** [comment]
I'll work on that shortly thank you for reminding me of the task 🙇

**Comment by hayata-suenaga** [comment]
Holding on [this issue (internal issue)](https://github.com/Expensify/Expensify/issues/333755)

**Comment by hayata-suenaga** [comment]
this has a low priority but I can get back to the internal issue (that is holding this issue) next week.

**Comment by hayata-suenaga** [comment]
I'm changing the priority to low as this doesn't affect the app at all

**Comment by dylanexpensify** [comment]
now that that's merged, can this be unblocked @hayata-suenaga?

**Comment by hayata-suenaga** [comment]
@dylanexpensify I'm still [debugging the issue here](https://github.com/Expensify/react-beautiful-dnd/actions/runs/7213616987)

**Comment by DylanDylann** [comment]
@hayata-suenaga Please pin me once we can continue this https://github.com/Expensify/App/issues/28797

**Comment by hayata-suenaga** [comment]
prioritizing higher priority issues

**Comment by hayata-suenaga** [comment]
[the fork](https://github.com/Expensify/react-beautiful-dnd/) was created but I have to setup a process to publish to npm

**Comment by DylanDylann** [comment]
@hayata-suenaga Any update here?

**Comment by hayata-suenaga** [comment]
might have time to work on this this month

**Comment by DylanDylann** [comment]
@hayata-suenaga Any update here?

**Comment by hayata-suenaga** [comment]
sorry this GH ticket has the least priority in my backlog

**Comment by hayata-suenaga** [comment]
same as above 🙇

**Comment by hayata-suenaga** [comment]
not a high priority issue now. will come back to this when I have time

**Comment by hayata-suenaga** [comment]
will come back to this when I have time 🙇

**Comment by DylanDylann** [comment]
@rafecolton Any update here?

**Comment by rafecolton** [comment]
Sorry I'm not involved in this particular issue, I just removed Hayata's assignment. Please ask in Slack.

**Comment by trjExpensify** [comment]
@DylanDylann what exactly needs to be done here? I can see we have the fork of this library, what are we waiting on?

**Comment by DylanDylann** [comment]
Waiting for @hayata-suenaga to fix this problem https://github.com/Expensify/App/issues/28797#issuecomment-1858272627

**Comment by DylanDylann** [comment]
@hayata-suenaga Could you help to give an ETA on this issue? Or should we close this issue?

cc @trjExpensify

**Comment by trjExpensify** [comment]
@DylanDylann, Hayata is no longer at Expensify. I think we close it for focus and circle back if we have a need for it.

**Comment by DylanDylann** [comment]
@trjExpensify Yes

**Comment by trjExpensify** [comment]
Cool, done.

**Comment by DylanDylann** [comment]
@trjExpensify In this issue, I created two PRs and one of them was merged 
- Merged:  https://github.com/Expensify/react-beautiful-dnd/pull/1
- Created (Not merged yet): https://github.com/Expensify/App/pull/29875

As mentioned on [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1723055151068559?thread_ts=1723015261.405329&cid=C02NK2DQWUX), the contributor should get fully paid because I am hired and did the work

I am happy to hear your though. Thanks

**Comment by DylanDylann** [comment]
It has been almost a year since I started working here

**Comment by trjExpensify** [comment]
Oh sorry, I thought you were settled on that. Cool, sent you an offer for $500 per the issue title.

**Comment by DylanDylann** [comment]
@trjExpensify Thank you! I accepted the offer

**Comment by trjExpensify** [comment]
Paid!
